packages/runtime/src/templates/getHandler.ts

```diff
@@ -176,12 +176,19 @@ const makeHandler = ({ conf, app, pageRoot, NextServer, staticManifest = [], mod
           // Only cache for 60s if default TTL provided
           result.ttl = 60
         }
-        if (result.ttl > 0) {
-          requestMode = `odb ttl=${result.ttl}`
-        }
       }
       multiValueHeaders['cache-control'] = ['public, max-age=0, must-revalidate']
     }
+
+    // ISR 404s are not served with SWR headers so we need to set the TTL here
+    if (requestMode === 'odb' && result.statusCode === 404) {
```

Comment:
[dust] If we can never have a 404 with an `swr` header then we don't need the 404 in `ephemeralCodes` above.